main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -2336,6 +2337,16 @@ object Parser2 {
       close(mark, TreeKind.Expr.Do)
     }
 
+    private def doInvokeMethod2Expr()(implicit s: State): Mark.Closed = {
+      assert(at(TokenKind.Dot))
+      val mark = open()
+      expect(TokenKind.Dot, SyntacticContext.Expr.Dot)
+      name(NAME_QNAME, context = SyntacticContext.Expr.InvokeMethod2)
```

Comment:
I think you can just remove this line and the line below it.

Reply:
Lines 2344 and 2345 to be specific

Reply:
Thank you for all your solutions. I can see clearer now.
However, it seems like that with the current code it only parses one dot, so I added another expect TokenKind.Dot statement. In addition, expression instead of name seemed to not parse `x` only. I will keep trying some things before pushing, otherwise it seems to work now with above changes.